File: printer-cloud-new/services/types/decree.ts
export interface ShowDecreeAPIResponse {
  id: number;
  decreeNumber: number;
  decreeDate: string;
  decreeUrl: string;
  lawNumber: number | null;
  lawDate: string | null;
  lawUrl: string | null;
  body: string | null;
  organizationId: number;
  createdAt: string;
  updatedAt: string;
}

File: PrinterCloud/app/models/printer_cloud/decree.rb
module PrinterCloud
  class Decree < ApplicationRecord
    include Orderable

    self.table_name = 'printer_cloud.decrees'

    belongs_to :organization

    validates :organization_id, uniqueness: true
    validates :decree_number, :decree_date, :decree_url, presence: true
    validates :decree_date, comparison: { less_than_or_equal_to: Date.today }, if: -> { decree_date_changed? }
    validates :law_date, comparison: { less_than_or_equal_to: Date.today }, if: lambda {
                                                                                  law_date_changed? && law_date.present?
                                                                                }
    validates :body, length: { maximum: 450 }
  end
end
